data/trie/trieStorageManager.go

```diff
@@ -184,33 +184,37 @@ func (tsm *trieStorageManager) ExitSnapshotMode() {
 
 // Prune removes the given hash from db
 func (tsm *trieStorageManager) Prune(rootHash []byte) {
+	tsm.storageOperationMutex.Lock()
+	defer tsm.storageOperationMutex.Unlock()
+
 	log.Trace("trie storage manager prune", "root", rootHash)
 
 	if tsm.snapshotInProgress > 0 {
+		tsm.pruningBuffer.add(rootHash)
 		return
 	}
 
-	if tsm.snapshotsBuffer.contains(rootHash[:len(rootHash)-1]) {
+	oldHashes := tsm.pruningBuffer.removeAll()
+	oldHashes = append(oldHashes, rootHash)
+
+	for i := range oldHashes {
```

Comment:
I think this code should be called in ExistSnapshotMode as well - when snapshot buffer len is == 0.

Reply:
Done.